src/test/java/com/hillel/items_exchange/chat/ChatEntitiesIntegrationTest.java

```diff
@@ -0,0 +1,158 @@
+        Message message = createMessageWithoutAttachments(user, chat);
+        chat.getMessages().add(message);
+        chatRepository.saveAndFlush(chat);
+    }
+
+    @Test
+    @DataSet(value = {"chat/init_db.yml", "chat/add_user_to_existing_chat.yml"})
+    @ExpectedDataSet(value = {"chat/init_db.yml",
+            "chat/add_user_to_existing_chat.yml",
+            "chat/add_message_with_attachments_for_every_user_in_chat.yml"},
+            ignoreCols = {"created", "updated", "file_content"})
+    @ExportDataSet(format = DataSetFormat.YML, outputName = "target/exported/testing.yml")
+    public void shouldAddNewMessageWithTwoAttachmentsForAllUsersOfChat() {
+        Chat chat = getExistingChat();
+        chat.getUsers().forEach(user -> {
+            Message message = createMessageWithoutAttachments(user, chat);
+            message.setAttachments(List.of(
+                    createAttachment(message, "png"),
+                    createAttachment(message, "jpg")
+            ));
+            user.getMessages().add(message);
+            userRepository.saveAndFlush(user);
+        });
+    }
+
+    @Test
+    @DataSet(value = {"chat/init_db.yml",
+            "chat/add_new_chat_to_user.yml",
+            "chat/add_new_message_without_attachments_to_chat.yml"})
+    @ExpectedDataSet("chat/remove_user_and_chats.yml")
+    @ExportDataSet(format = DataSetFormat.YML, outputName = "target/exported/testing.yml")
+    public void shouldRemoveUserWithAllOfHisChatsAndMessagesWithAttachments() {
+        User user = getExistingUserByUsername(SECOND_USER_USERNAME);
+        userRepository.delete(user);
+        userRepository.flush();
+    }
+
+    @Test
+    @DataSet(value = {"chat/init_db.yml",
+            "chat/add_user_to_existing_chat.yml",
+            "chat/add_message_with_attachments_for_every_user_in_chat.yml"})
+    @ExpectedDataSet(value = "chat/remove_user_from_chat_with_more_than_one_user.yml",
+            ignoreCols = {"created", "updated", "file_content"})
+    @ExportDataSet(format = DataSetFormat.YML, outputName = "target/exported/testing.yml")
+    public void shouldRemoveUserWithHisMessagesAndAttachmentsForThemButLeaveChat() {
+        final User existingUser = getExistingUserByUsername(SECOND_USER_USERNAME);
+        existingUser.getChats().remove(0);
+        userRepository.delete(existingUser);
+        userRepository.flush();
+    }
+
+    @Test
+    @DataSet(value = {"chat/init_db.yml"})
+    @ExpectedDataSet("chat/add_user_to_blacklist.yml")
+    public void shouldAddUserToBlacklist() {
+        final User user1 = getExistingUserByUsername(FIRST_USER_USERNAME);
+        final User user2 = getExistingUserByUsername(SECOND_USER_USERNAME);
+        user1.getBlacklistedUsers().add(user2);
+        userRepository.saveAndFlush(user1);
+    }
+
+    private User getExistingUserByUsername(String username) {
+        return userRepository.findByUsername(username).orElseThrow(EntityNotFoundException::new);
+    }
+
```

Comment:
As far as I know, using helper methods in Test classes is bad practice, you can port them to TestUtil instead.